src/functions.rs

```diff
@@ -2009,12 +2099,20 @@ impl ValidatedCall {
         let cil_name = match &call.class_name {
             Some(class_name) => {
                 // Resolve aliases
-                match types.get(convert_class_name_if_this(class_name, parent_type)?.as_ref()) {
+                let tmp_str = match types
+                    .get(convert_class_name_if_this(class_name, parent_type)?.as_ref())
+                {
                     Some(type_name) => get_cil_name(Some(&type_name.name), &call.name),
                     None => call.get_cil_name(), // Expected to error out below
+                };
+                // If we have a cast name we really want to call that instead.
+                if let Some(cast_name) = &call.cast_name {
+                    tmp_str.replace(&class_name.to_string(), cast_name.to_string().as_ref())
```

Comment:
Could this cause a problem if the function name contains the class name as a substring?  In other words, if I have a resource named foo, with a function named foobar().  Then the cil name is foo-foobar(), and if I cast to baz, this replace, would turn it into the invalid baz-bazbar() instead of the intended baz-foobar().